### PR TITLE
New seralizer logic

### PR DESCRIPTION
**Pull request by cleptric**
Based on https://github.com/getsentry/sentry-php/pull/1612

This declutters the `PayloadSeralizer` and adds distinct `EnvelopeItem` types based on the event type.
While this is not strictly DRY, transactions will soon diverge heavily from errors, hence already laying the groundwork.

### CODE REVIEW

File: src/Serializer/EnvelopItems/EventItem.php [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+        if ($event->getStartTimestamp() !== null) {
+            $payload['start_timestamp'] = $event->getStartTimestamp();
+        }
+
+        if ($event->getLevel() !== null) {
+            $payload['level'] = (string) $event->getLevel();
+        }
+
+        // TODO(michi) remove
```
Comment (stephanie-anderson) [inline src/Serializer/EnvelopItems/EventItem.php:44]: @cleptric 👋

File: src/Serializer/EnvelopItems/TransactionItem.php [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+        if (!empty($event->getBreadcrumbs())) {
+            $payload['breadcrumbs']['values'] = array_map([self::class, 'serializeBreadcrumb'], $event->getBreadcrumbs());
+        }
+
+        if (!empty($event->getRequest())) {
+            $payload['request'] = $event->getRequest();
+        }
+
+        // TODO(michi) clarify
```
Comment (stephanie-anderson) [inline src/Serializer/EnvelopItems/TransactionItem.php:122]: @cleptric more to remove I guess

File: src/Serializer/EnvelopItems/TransactionItem.php [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+        if ($event->getStartTimestamp() !== null) {
+            $payload['start_timestamp'] = $event->getStartTimestamp();
+        }
+
+        if ($event->getLevel() !== null) {
+            $payload['level'] = (string) $event->getLevel();
+        }
+
+        // TODO(michi) remove
```
Comment (stephanie-anderson) [inline src/Serializer/EnvelopItems/TransactionItem.php:44]: @cleptric

**Review by stephanie-anderson — COMMENTED** [review]
Found some comments that look like they should be removed. Please have a look, otherwise seems fine

**Comment by cleptric** [comment]
All cleaned up

**Review by stephanie-anderson — APPROVED** [review]

**Pull request merged**